src/librbd/internal.cc

```diff
@@ -2551,12 +2550,14 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force,
           }
         }
       }
-    }
 
-    r = mirror_image_disable_internal(ictx, force);
-    if (r < 0) {
-      return r;
+      r = mirror_image_disable_internal(ictx, force);
```

Comment:
sorry, did not notice the lock :(
updated and moved the code needs the `snap_lock` into a block.
thanks.